### PR TITLE
tests(*) proxy stress testing

### PR DESCRIPTION
**Pull request by locao**
This PR stress tests the balancer's proxying feature.

(please do not merge this PR while Kong/kong-build-tools/pull/375 is not merged)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -99,7 +99,7 @@ setup-kong-build-tools:
 	-rm -rf $(KONG_BUILD_TOOLS_LOCATION)
 	-git clone https://github.com/Kong/kong-build-tools.git $(KONG_BUILD_TOOLS_LOCATION)
 	cd $(KONG_BUILD_TOOLS_LOCATION); \
-	git reset --hard $(KONG_BUILD_TOOLS); \
+	git reset --hard && git checkout $(KONG_BUILD_TOOLS); \
```
Comment (kikito) [inline Makefile:102]: Why is this needed?

**Comment by kikito** [comment]
I think this is ok, but I think I'd rather have Vegeta installed as part of `make dev` instead of as a step in `kong-build-tools`.

File: Makefile (hunk 1 above)
Comment (locao) [reply Makefile:102]: git can't reset to branches, only to tags. With this change it's possible to checkout an unreleased kong-build-tools version.

Comment (hutchic) [reply Makefile:102]: I thought it would work if you prefixed with `origin/` but maybe I'm wrong

**Comment by dndx** [comment]
@locao I merged https://github.com/Kong/kong-build-tools/pull/375, so you can get rid of the custom beaches from the change now :)

**Comment by dndx** [comment]
@locao seems that the base branch is broken after the latest push, now it has more than 30 commits in the PR.

**Review by dndx — APPROVED** [review]

**Review by dndx — APPROVED** [review]
